app/controllers/claims/schools/users_controller.rb

```diff
@@ -1,3 +1,40 @@
 class Claims::Schools::UsersController < ApplicationController
   include Claims::BelongsToSchool
+
+  def index
+    @users = @school.users
```

Comment:
Sorry to open this again, I think we have scoping issue. If you know the ID of a school of a different user you can go to that user's school. I'm not saying we should fix this in this PR as I expect we have similar scoping issues in other places. But to me this feels like where policies come in.

I would expect an index policy here that checks the school_id in the param against the current user schools. Something like:
```
School policy 

def index
  current_user.schools.pluck(:id).inlcude?(params.require(:school_id))
end
```


https://github.com/DFE-Digital/itt-mentor-services/assets/11318084/a0f5a973-363a-43a4-9d09-43e7f2cb9765

Don't want to block this PR, but we should probably address this either here or in a different PR

Reply:
Thinking about it more, we can also do this scoping in the controller, we just need to handle the user not being allowed to view another school gracefully. Don't mind how we solve this.

Reply:
@CatalinVoineag I like the idea of adding Pundit in, but maybe only when we have resource-based permissions in place. Right now, any user belonging to a school can edit any other user, claim, etc. For now, we can likely get by by scoping down the lookup of school.

@gms-gs 
We could update the `Claims::BelongsToSchool` to scope down the top-level scope to `current_user.schools.find_by(params.require(:id))`. The challenge here is to now make sure it's using the right namespace.

```diff
class Claims::User < User
  default_scope { claims }

+ has_many :schools,
+          through: :memberships,
+          source: :organisation,
+          source_type: "School"
end
```

Adding this in should make sure `current_user.schools` uses the `Claims::School` model as long as `current_user` is of class `Claims::User`. As far as I can tell, the codebase currently does this namespacing.

Reply:
Please just be aware that `Claims::BelongsToSchool` is also used by the support controllers.

As a rule, support users will need access to _all_ schools. But it seems wasteful & like unnecessary overhead to set up membership records linked every support user with every school.

Instead, I think you'll probably want a `Claims::SupportUser` which redefines the association to `has_many :schools, -> { claims }` so every school is available on `current_user`.